str/runners/str_iterative_gangstr_runner.py

```diff
@@ -37,63 +34,65 @@
-    )
-    b = hb.Batch(backend=backend, default_image=os.getenv('DRIVER_IMAGE'))
+    b = get_batch()
 
-    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(dataset, list(external_wgs_ids))
-    cpg_id_to_external_id = {cpg_id: external_wgs_id for external_wgs_id, cpg_id in external_id_to_cpg_id.items()}
-    if dataset in ['tob-wgs', 'hgdp']:
+    if any(keyword in dataset for keyword in ['hgdp', 'thousand-genomes']):
         ref_fasta = 'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
     else:
-        ref_fasta = reference_path('broad/ref_fasta')
-    if dataset == 'tob-wgs':
-        analysis_query_model = AnalysisQueryModel(
-            sample_ids=list(external_id_to_cpg_id.values()),
-            projects=[dataset],
-            type=AnalysisType('cram'),
-            status=AnalysisStatus('completed'),
-            meta={'sequencing_type': 'genome', 'source': 'nagim'},
-        )
-    else:
-        analysis_query_model = AnalysisQueryModel(
-            sample_ids=list(external_id_to_cpg_id.values()),
-            projects=[dataset],
-            type=AnalysisType('cram'),
-            status=AnalysisStatus('completed'),
-            meta={'sequencing_type': 'genome'},
-        )
-    crams_path = AnalysisApi().query_analyses(analysis_query_model)
-    cpg_sids_with_crams = set(sid for sids in crams_path for sid in sids['sample_ids'])
-    cpg_sids_without_crams = set(cpg_id_to_external_id.keys()) - cpg_sids_with_crams
-    if cpg_sids_without_crams:
-        external_wgs_sids_without_crams = ', '.join(cpg_id_to_external_id[sid] for sid in cpg_sids_without_crams)
-        logging.warning(f'There were some samples without CRAMs: {external_wgs_sids_without_crams}')
+        ref_fasta = str(reference_path('broad/ref_fasta'))
+
     gangstr_regions = b.read_input(variant_catalog)
 
+    # Working with CRAM files requires the reference fasta
+    ref = b.read_input_group(
+        **dict(
+            base=ref_fasta,
+            fai=ref_fasta + '.fai',
+            dict=ref_fasta.replace('.fasta', '').replace('.fna', '').replace('.fa', '') + '.dict',
+        ),
+    )
+
     # Iterate over each sample to call GangSTR
-    for cram_obj in crams_path:
+    for cpg_id in cpg_ids:
+        # retrieve corresponding cram path
+        cram_retrieval_query = gql(
+            """
+            query MyQuery($dataset: String!,$cpg_id: [String!]!) {
+            project(name: $dataset) {
+            sequencingGroups(id: {in_: $cpg_id}) {
+            id
+            sample {
+                externalId
+            }
+            analyses(type: {eq: "cram"}, active: {eq: true}) {
+                output
+                timestampCompleted
+            }
+            }
+        }
+
+        }
+            """,
+        )
+        response = query(
+            cram_retrieval_query,
+            variables={'dataset': dataset, 'cpg_id': cpg_id},
+        )
```

Comment:
You'd be better off here parametrising this query to take all your CPG IDs, and querying once

Reply:
I could...but it works....and this script will only be used once. Can I not make this change :)